lxd/storage_volumes_config.go

```diff
@@ -89,8 +85,15 @@ func storageVolumeValidateConfig(name string, config map[string]string, parentPo
 
 func storageVolumeFillDefault(name string, config map[string]string, parentPool *api.StoragePool) error {
 	if parentPool.Driver == "dir" {
-		config["size"] = "0"
+		config["size"] = ""
 	} else if parentPool.Driver == "lvm" {
+		if config["block.filesystem"] == "" {
```

Comment:
`block.filesystem` is set to the pool value `volume.block.filesystem` or receives `ext4` as default value. In any case, it will always be set on the `volume` level but will only appear on the pool level if explicitly set to something by the user.